packages/lib/shared/utils/errors.ts

```diff
@@ -144,8 +155,21 @@ function parseRequestError(error: Error, chainId: number): TransactionConfig | u
       return
     }
   }
+}
 
-  return
+function parseRawCallArgumentsError(error: Error, chainId: number): TransactionConfig | undefined {
+  const toMatch = error.message.match(/to:\s*([^\n]+)/)
+  const dataMatch = error.message.match(/data:\s*([^\n]+)/)
+
+  const to = toMatch?.[1].trim() ?? ''
+  const data = dataMatch?.[1].trim() ?? ''
+
+  return {
+    data: data as Hex,
+    to: to as Address,
+    chainId,
+    account: '0x0000000000000000000000000000000000000000',
```

Comment:
Is this not a bit misleading? Should it not be in a specific utils file for tests? Otherwise, should the user's address not be parsed and used?